stonesoup/sensor/sensor.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC
+
+from ..platform.simple import FixedSensorPlatform
+from .base import BaseSensor
+from ..types.array import StateVector
+from ..types.state import State
+
+
+class Sensor(BaseSensor, ABC):
```

Comment:
Why not remove `BaseSensor` and just have this?

Reply:
Because of a circular reference: This needs to know about the `FixedSensorPlatform` class, to build one as a default internal platform. All `SensorPlatforms` need to know about the basic `BaseSensor` functionality to know how to handle them. In order to resolve this I had to split the functionality into one class that can be imported by `platform.simple` and one that imports `platform.simple`. I tried first to make `BaseSensor` private (by calling it `_Sensor`) but that means you lose the documentation for it (unless you know a way of fixing that).

Reply:
Circular references are a right pain 😩